### ISSUE TITLE
Turning off deadeye still causes desyncs in multiplayer

### ISSUE THREAD

**Issue by Lesamuen** [SolastaMods/SolastaCommunityExpansion#1373]
Related to #1340. This time, I got the log. This is on version 1F, but I don't think the related code has been changed for 1G.

The relevant log section is as follows:

```
[SolastaCommunityExpansion] AttackMain
[SolastaCommunityExpansion] HandleCharacterAttack
[SolastaCommunityExpansion] HandleCharacterAttackDamage
NullReferenceException: Object reference not set to an instance of an object
  at SRDebugger.Services.Implementation.DebugTriggerImpl.OnError (SRDebugger.Services.IConsoleService console) [0x00000] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at SRDebugger.Services.Implementation.StandardConsoleService.AdjustCounter (UnityEngine.LogType type, System.Int32 amount) [0x00031] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at SRDebugger.Services.Implementation.StandardConsoleService.UnityLogCallback (System.String condition, System.String stackTrace, UnityEngine.LogType type) [0x0SolastaCommunityExpansion] AttackMain
[SolastaCommunityExpansion] HandleCharacterAttack
[SolastaCommunityExpansion] HandleCharacterAttackDamage
NullReferenceException: Object reference not set to an instance of an object
  at SRDebugger.Services.Implementation.DebugTriggerImpl.OnError (SRDebugger.Services.IConsoleService console) [0x00000] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at SRDebugger.Services.Implementation.StandardConsoleService.AdjustCounter (UnityEngine.LogType type, System.Int32 amount) [0x00031] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at SRDebugger.Services.Implementation.StandardConsoleService.UnityLogCallback (System.String condition, System.String stackTrace, UnityEngine.LogType type) [0x00043] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at UnityEngine.Application.CallLogCallback (System.String logString, System.String stackTrace, UnityEngine.LogType type, System.Boolean invokedOnMainThread) [0x00030] in <309a05e2dd0a4e38aa2f0404a7fd287c>:0 
UnityEngine.DebugLogHandler:Intern0043] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at UnityEngine.Application.CallLogCallback (System.String logString, System.String stackTrace, UnityEngine.LogType type, System.Boolean invokedOnMainThread) [0x00030] in <309a05e2dd0a4e38aa2f0404a7fd287c>:0 
UnityEngine.DebugLogHandler:Internal_Log(LogType, LogOption, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
Trace:LogError(String)
<ExecuteImpl>d__45:MoveNext()
SolastaCommunityExpansion.Patches.CustomFeatures.CustomSpells.<Postfix>d__1:MoveNext()
TA.Coroutine:Run()
GameLocationActionManager:UpdateImpl()
GameLocationActionManager:FixedUpdate()
 
(Filename: <a49d12eccc334faeb2cb9b4bf638c070> Line: 0)

Error Context : Name = TurnOffDeadeyePower, SourceGuid = 119, Level = 0, ActionType = Main, SourceType = Power, EffectSourceType = Power, SaveDC = 12, SourceAbility = Intelligence, SchoolOfMagic = , MetamagicOption =  
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 39)

Uploading Crash Report
NullReferenceException: Object reference not set to an instance of an object
  at SRDebugger.Services.Implementation.DebugTriggerImpl.OnError (SRDebugger.Services.IConsoleService console) [0x00000] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at SRDebugger.Services.Implementation.StandardConsoleService.AdjustCounter (UnityEngine.LogType type, System.Int32 amount) [0x00031] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at SRDebugger.Services.Implementation.StandardConsoleService.UnityLogCallback (System.String condition, System.String stackTrace, UnityEngine.LogType type) [0x00043] in <a49d12eccc334faeb2cb9b4bf638c070>:0 
  at UnityEngine.Application.CallLogCallback (System.String logString, System.String stackTrace, UnityEngine.LogType type, System.Boolean invokedOnMainThread) [0x00030] in <309a05e2dd0a4e38aa2f0404a7fd287c>:0 
UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
Trace:LogException(Exception)
<ExecuteImpl>d__45:MoveNext()
SolastaCommunityExpansion.Patches.CustomFeatures.CustomSpells.<Postfix>d__1:MoveNext()
TA.Coroutine:Run()
GameLocationActionManager:UpdateImpl()
GameLocationActionManager:FixedUpdate()
 
(Filename: <a49d12eccc334faeb2cb9b4bf638c070> Line: 0)

Exception: [TACTICAL INVISIBLE FOR PLAYERS] null GameLocationEffect in CharacterActionMagicEffect.ExecuteImpl().

UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
Trace:LogException(Exception)
<ExecuteImpl>d__45:MoveNext()
SolastaCommunityExpansion.Patches.CustomFeatures.CustomSpells.<Postfix>d__1:MoveNext()
TA.Coroutine:Run()
GameLocationActionManager:UpdateImpl()
GameLocationActionManager:FixedUpdate()
 
(Filename:  Line: -1)

[SolastaCommunityExpansion] PowerNoCost
[SolastaCommunityExpansion] AttackMain
```

**Comment by ThyWoof** [comment]
Can you paste the full log file as a l
Download link? I need the context before the error.

**Comment by Lesamuen** [comment]
Deadeye works just fine as long as you don't turn it off, but the moment you turn it off you get the "SOLASTA HAS DETECTED AN INSTABILITY" message and deadeye doesn't actually turn off for the other clients still.

This log was generated by loading a save right before a battle, shooting with the character with deadeye to initiate the battle, and then having him try to turn off deadeye.

[Player.log](https://github.com/SolastaMods/SolastaCommunityExpansion/files/8865555/Player.log)

**Comment by ThyWoof** [comment]
Does it also happen with power attack or only deadeye? I believe I found a fix but need to confirm this first

**Comment by Lesamuen** [comment]
He's a level 4 character, only has deadeye.

**Comment by ThyWoof** [comment]
Thanks for all the details. Fixed on 1.3.57.2B
